File: TelerikUIExamples/TelerikUIExamples/LogarithmicAxis.m
//
//  LogarithmicAxis.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import "LogarithmicAxis.h"
#import <TelerikUI/TelerikUI.h>

@implementation LogarithmicAxis

- (void)viewDidLoad {
    [super viewDidLoad];
    
    TKChart *chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    chart.legend.hidden = NO;
    [self.view addSubview:chart];
    
    TKDataSource *dataSource = [[TKDataSource alloc] initWithDataFromJSONResource:@"electricity" ofType:@"json" rootItemKeyPath:@"data"];
    [dataSource sortWithKey:@"year" ascending:YES];

    dataSource.itemSource = @[ [[TKDataSourceGroup alloc] initWithItems:dataSource.items valueKey:@"nuclear" displayKey:@"year"],
                               [[TKDataSourceGroup alloc] initWithItems:dataSource.items valueKey:@"hydro" displayKey:@"year"],
                               [[TKDataSourceGroup alloc] initWithItems:dataSource.items valueKey:@"solar" displayKey:@"year"],
                             ];

    NSArray *colors = @[ [UIColor colorWithRed:0.318f green:0.384f blue:0.737f alpha:1.00f],
                         [UIColor colorWithRed:0.039f green:0.631f blue:0.933f alpha:1.00f],
                         [UIColor colorWithRed:0.271f green:0.678f blue:0.373f alpha:1.00f],
                       ];
    
    [dataSource.settings.chart createSeries:^TKChartSeries *(TKDataSourceGroup *group) {
        TKChartAreaSeries *series = [TKChartAreaSeries new];
        series.title = [group.valueKey capitalizedString];
        series.style.fill = [TKSolidFill solidFillWithColor:colors[[dataSource.items indexOfObject:group]]];
        return series;
    }];
    
    chart.yAxis = [TKChartLogarithmicAxis new];
    chart.dataSource = dataSource;
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/LogarithmicAxis.swift
//
//  LogarithmicAxis.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//
//

class LogarithmicAxis: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        chart.legend.hidden = false;
        self.view.addSubview(chart)
        
        let datasource = TKDataSource(dataFromJSONResource: "electricity", ofType: ".json", rootItemKeyPath: "data")
        datasource.sortWithKey("year", ascending: true)
        
        datasource.itemSource = [TKDataSourceGroup(items: datasource.items, valueKey: "nuclear", displayKey: "year"),
            TKDataSourceGroup(items: datasource.items, valueKey: "hydro", displayKey: "year"),
            TKDataSourceGroup(items: datasource.items, valueKey: "solar", displayKey: "year")
        ];
        
        var colors = [UIColor(red: 0.318, green: 0.384, blue: 0.737, alpha: 1.00),
        UIColor(red: 0.039, green: 0.631, blue: 0.933, alpha: 1.00),
        UIColor(red: 0.271, green: 0.678, blue: 0.373, alpha: 1.00)]
        
        datasource.settings.chart.createSeries { (TKDataSourceGroup group) -> TKChartSeries! in
            let series = TKChartAreaSeries()
            series.title = group!.valueKey!.capitalizedString
            series.style.fill = TKSolidFill(color:colors[datasource.itemSource!.indexOfObject(group!)])
            return series
        }
        
        chart.yAxis = TKChartLogarithmicAxis()
        chart.dataSource = datasource
    }
}
